streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -364,7 +390,7 @@ private void materialize(KTableSource<K, ?> source) {
         synchronized (source) {
             if (!source.isMaterialized()) {
                 StateStoreSupplier storeSupplier =
-                        new KTableStoreSupplier<>(source.topic, keySerde, valSerde, null);
+                        new KTableStoreSupplier<>(storeName, keySerde, valSerde, null);
```

Comment:
See my previous comment above: we should set the corresponding `KTableImpl` of the `KTableSource`'s state store name for the `getStoreName` call.